### DISCUSSION TITLE
Unit tests in the project does not run tests in `src/test` directory

### DISCUSSION THREAD

**Discussion by Xayer** [mswjs/msw#1197] [Q&A]
### Scope

Improves an existing behavior

### Compatibility

- [ ] This is a breaking change

### Feature description

I was checking out the source code, as I wanted to implement a new feature.

I ran the unit tests, and discovered that none of the tests from `src/test` were included.

This is because they are not included in the `roots`part of the jest.config.js file.

``` javascript
// jest.config.js
module.exports = {
  roots: ['<rootDir>/src', '<rootDir>/cli'], // no '<rootDir>/test entry is present.
```

After adding the directory to the `roots`, I experienced alot of failing tests.
I would like someone to verify this, as It might only be onmy environment.

**Comment by kettanaito** [comment]
Hey, @Xayer.

Could you please include the path to the test you've added and trying to run?

`<rootDir>/src` covers `src/test` and any other tests located under `src`. We use those for unit testing. I also recommend you to read the [Testing Guidelines](https://github.com/mswjs/msw/blob/main/CONTRIBUTING.md#tests).

**Reply by Xayer** [reply]
I made a mistake when I published this issue. The `test` directory is **not** located inside of the `src` directory.

![image](https://user-images.githubusercontent.com/1837767/162045360-d405c969-28a0-47eb-af9d-4d6817af243b.png)

That is why I felt the need to add the directory to the jest.config.js directory.

**Reply by kettanaito** [reply]
That's expected. We write integration tests in `./test`. You can run them via `yarn test:integration` command. I believe this is explained in the contributing guidelines I've shared.

**Comment by kettanaito** [comment]
Also, I'd encourage you to open a discussion around the feature you'd like to propose before developing it. I am thankful for your effort, that's why I feel it's crucial to discuss things before spending time on them.

**Reply by Xayer** [reply]
I wanted to test the waters with [a tweet](https://mobile.twitter.com/Xayer96/status/1511614848865652736).
Sounds like you've wanted to do it for a while. Should I create a separate discussion thread for it, or does one already exist? 

Edit: I've found the existing discussion: https://github.com/mswjs/msw/discussions/831#discussioncomment-2405680

**Reply by kettanaito** [reply]
Oh, nice! Yeah, it's been great to have `.listHandlers()` instead of `.printHandlers()` so people could wrap the list of handlers in `console.log` if they wish for introspection.
